### PR TITLE
[amp] dygraph amp support param_group

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

Others
### Describe

- dygraph amp support param_group

```
linear_1 = paddle.nn.Linear(10, 10)
linear_2 = paddle.nn.Linear(10, 10)
inp = paddle.uniform(shape=[10, 10], min=-0.1, max=0.1)
with paddle.amp.auto_cast():
    out = linear_1(inp)
    out = linear_2(out)
loss = paddle.mean(out)
sgd = paddle.optimizer.SGD(
    learning_rate=0.1,
    parameters=[{
        'params': linear_1.parameters()
    }, {
        'params': linear_2.parameters(),
        'weight_decay': 0.001,
        'learning_rate': 0.1
    }],
    weight_decay=0.01)              
scaler=paddle.amp.GradScaler()     
scaler.scale(out).backward()
scaler.step(sgd)
sgd.clear_grad()
```
- add `step` method for `class GradScaler`, to be consistent with `class Optimizer`, since `Optimizer.minimize()` does not support `param_group`

### CODE REVIEW

**Review by jerrywgz — COMMENTED** [review]
LGTM

File: python/paddle/amp/grad_scaler.py [hunk 1]
```diff
@@ -146,6 +146,45 @@ def minimize(self, optimizer, *args, **kwargs):
...
+        """
+        This function is similar as `optimizer.step()`, which performs parameters updating.
+        
+        If the scaled gradients of parameters contains NAN or INF, the parameters updating is skipped.
+        Otherwise, it first unscales the scaled gradients of parameters, then updates the parameters.
+        Args:
+            optimizer(Optimizer):  The optimizer used to update parameters.
+        Examples:
+            .. code-block:: python
+                import paddle
```
Comment (TCChenlong) [inline python/paddle/amp/grad_scaler.py:163]: import 前要加一个空行

File: python/paddle/amp/grad_scaler.py [hunk 2]
```diff
@@ -146,6 +146,45 @@ def minimize(self, optimizer, *args, **kwargs):
         """
         return super(GradScaler, self).minimize(optimizer, *args, **kwargs)
 
+    def step(self, optimizer):
+        """
+        This function is similar as `optimizer.step()`, which performs parameters updating.
+        
+        If the scaled gradients of parameters contains NAN or INF, the parameters updating is skipped.
+        Otherwise, it first unscales the scaled gradients of parameters, then updates the parameters.
+        Args:
```
Comment (TCChenlong) [inline python/paddle/amp/grad_scaler.py:156]: Args 前要加空行

File: python/paddle/amp/grad_scaler.py [hunk 3]
```diff
@@ -146,6 +146,45 @@ def minimize(self, optimizer, *args, **kwargs):
...
+            optimizer(Optimizer):  The optimizer used to update parameters.
+        Examples:
```
Comment (TCChenlong) [inline python/paddle/amp/grad_scaler.py:159]: Examples 前要加空行

**Review by phlrain — DISMISSED** [review]

File: python/paddle/amp/grad_scaler.py (hunk 1 above)
Comment (zhiqiu) [reply python/paddle/amp/grad_scaler.py:163]: done, thx

File: python/paddle/amp/grad_scaler.py (hunk 2 above)
Comment (zhiqiu) [reply python/paddle/amp/grad_scaler.py:156]: done, thx

File: python/paddle/amp/grad_scaler.py (hunk 3 above)
Comment (zhiqiu) [reply python/paddle/amp/grad_scaler.py:159]: done, thx

**Review by TCChenlong — APPROVED** [review]
LGTM for API docs

**Review by phlrain — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**